### PR TITLE
feat(SQLAdminFetcher): Support base `AuthClient` for `auth` and `loginAuth`

### LINKED ISSUE #200: Add support for specifying custom credentials
## Question

Getting this error 
```
Not authorized to access resource. Possibly missing permission cloudsql.instances.get on resource
```
I am connecting with my service account and I verified that the user has cloudsql.admin. Any help?

### PR DESCRIPTION
**Pull request by danielbankhead**
## Change Description

Adds custom auth support for `auth` and `loginAuth`.

## Checklist

- [x] Make sure to open an issue as a
  [bug/issue](https://github.com/GoogleCloudPlatform//issues/new/choose)
  before writing your code! That way we can discuss the change, evaluate
  designs, and agree on the general idea.
- [x] Ensure the tests and linter pass
- [x] Appropriate documentation is updated (if necessary)

## Relevant issues:

- Fixes #200

### CODE REVIEW

File: src/sqladmin-fetcher.ts [hunk 1]
```diff
@@ -69,21 +69,32 @@ function cleanGaxiosConfig() {
...
+      auth = new GoogleAuth({
+        authClient: loginAuth, // either an `AuthClient` or undefined
+        scopes: ['https://www.googleapis.com/auth/sqlservice.admin'],
+      });
+    }
+
     this.client = new Sqladmin({
       rootUrl: sqlAdminAPIEndpoint,
-      auth,
+      auth: auth as GoogleAuth,
```
Comment (danielbankhead) [inline src/sqladmin-fetcher.ts:97]: Requires the following to be resolved in order to remove the casting:
- https://github.com/googleapis/google-api-nodejs-client/issues/3357

File: test/sqladmin-fetcher.ts [hunk 2]
```diff
@@ -71,6 +71,28 @@ const mockRequest = (
     });
 };
 
+t.test('constructor', async t => {
```
Comment (danielbankhead) [inline test/sqladmin-fetcher.ts:74]: These tests aren't really 'tested' as this suite may be refactored shortly; they only exist for coverage purposes.

File: src/sqladmin-fetcher.ts [hunk 3]
```diff
@@ -69,21 +69,32 @@ function cleanGaxiosConfig() {
 }
 
 export interface SQLAdminFetcherOptions {
-  loginAuth?: GoogleAuth;
+  auth?: GoogleAuth<AuthClient> | AuthClient;
```
Comment (edosrecki) [inline src/sqladmin-fetcher.ts:72]: This property is unused, only `loginAuth` is used in the constructor.

**Review by edosrecki — CHANGES_REQUESTED** [review]

File: src/sqladmin-fetcher.ts (hunk 3 above)
Comment (ruyadorno) [reply src/sqladmin-fetcher.ts:72]: agreed, the constructor only needs a single `loginAuth` object that the other instances should extend from using `authClient`

**Review by ruyadorno — CHANGES_REQUESTED** [review]
We'll need to expose a public API (similar to https://github.com/GoogleCloudPlatform/cloud-sql-nodejs-connector/commit/7b6f6c1a1aef10480293002b3e0d15ab997d1e5e) along with README docs on how to use it in order to really fix #200

File: README.md [hunk 4]
```diff
@@ -20,12 +20,12 @@ alternative to the [Cloud SQL Auth Proxy](https://cloud.google.com/sql/docs/mysq
 while providing the following benefits:
 
 - **IAM Authorization:** uses IAM permissions to control who/what can connect to
-your Cloud SQL instances
+  your Cloud SQL instances
```
Comment (danielbankhead) [inline README.md:23]: Prettier-formatted. Line 324 is the relevant bit.

**Review by ruyadorno — APPROVED** [review]

**Pull request merged**